### PR TITLE
Update DG model and delete feature

### LINKED ISSUE #54: Add Delete Feature in Developer Guide and Update Model Diagrams
- Add Delete feature in developer guide
- Update DeleteSequenceDiagram
- Update ModelClassDiagram and BetterModelClassDiagram

### PR DESCRIPTION
**Pull request by shaokiat**
- Add Delete feature in developer guide
- Update DeleteSequenceDiagram
- Update ModelClassDiagram and BetterModelClassDiagram

### CODE REVIEW

**Review by mhdsyfq77 — CHANGES_REQUESTED** [review]
generally well done bro 👍🏽 

just one thing i noticed, doesn't our delete command parse an index rather than the module code? or did we decide we are going to change that and i totally forgot LOL

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -179,6 +179,39 @@ The following sequence diagram shows how the edit command is executed.
...
+
+Step 3. `Logic.execute()` then calls the `parseCommand`  method of the `gradPadParser` class to parse the string input.
+
+Step 4. `gradPadParser.parseCommand()` sees that this is an delete command, and so uses the `DeleteCommandParser`
+class to create a corresponding `DeleteCommand`.
+
+Step 5. In `DeleteCommandParser`, the ModuleCode is first extracted from the string input. 
+
+Step 6. A `DeleteCommand` is then created with the ModuleCode, and is passed back to the
+`LogicManager` in step 2.
```
Comment (yan-soon) [inline docs/DeveloperGuide.md]: I think same issue as the one I had, should we explicitly state the call of the DeleteCommandParser.parse() method?

File: docs/diagrams/DeleteSequenceDiagram.puml [hunk 2]
```diff
@@ -13,42 +14,48 @@ box Model MODEL_COLOR_T1
...
 end box
 
-[-> LogicManager : execute("delete 1")
+[-> LogicManager : execute("delete c/CS2103T")
 activate LogicManager
 
-LogicManager -> AddressBookParser : parseCommand("delete 1")
-activate AddressBookParser
+LogicManager -> GradPadParser : parseCommand("delete c/CS2103T")
+activate GradPadParser
```
Comment (yan-soon) [inline docs/diagrams/DeleteSequenceDiagram.puml]: I think our current delete system still works with indexes, but possibly can KIV for future updates?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -471,19 +504,19 @@ testers are expected to do more *exploratory* testing.
...
-1. Deleting a person while all persons are being shown
+1. Deleting a module.
 
-   1. Prerequisites: List all persons using the `list` command. Multiple persons in the list.
+   1. Prerequisites: Multiple modules in the 'Current Modules'. e.g. CS2103T in 'Current Modules'.
 
-   1. Test case: `delete 1`<br>
-      Expected: First contact is deleted from the list. Details of the deleted contact shown in the status message. Timestamp in the status bar is updated.
+   1. Test case: `delete CS2103T`<br>
+      Expected: CS2103T module is deleted from 'Current Modules'. Details of the deleted module shown in the status message. Timestamp in the status bar is updated.
```
Comment (Silvernitro) [inline docs/DeveloperGuide.md]: What does the timestamp in the status bar refer to ah?

**Review by Silvernitro — COMMENTED** [review]
Mostly LGTM!

Also I think there's a duplicate `ModelClassDiagram.png` in the root directory 🤔

File: docs/diagrams/DeleteSequenceDiagram.puml (hunk 2 above)
Comment (shaokiat) [reply docs/diagrams/DeleteSequenceDiagram.puml]: Yup for now it still works with indexes but I think we are planning to change it to ModuleCode in the near future so I just change the documentation first.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (shaokiat) [reply docs/DeveloperGuide.md]: I think it has got to do with the JavaFX status bar but I don't see it in the GUI

**Review by Silvernitro — APPROVED** [review]

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (shaokiat) [reply docs/DeveloperGuide.md]: Okay I'll remove it since its not needed.

**Review by mhdsyfq77 — APPROVED** [review]

**Pull request merged**